app/com/lynxanalytics/biggraph/graph_operations/NetworKit.scala

```diff
@@ -16,48 +16,67 @@ object NetworKitCommon {
       case v: Boolean => json.Json.toJson(v)
     }.toSeq)
   }
+  class WeightedGraphInput(weighted: Boolean) extends MagicInputSignature {
+    val vs = vertexSet
+    val es = edgeBundle(vs, vs)
+    val weight = if (weighted) edgeAttribute[Double](es) else null
+  }
 }
 
 object NetworKitComputeAttribute extends OpFromJson {
   def fromJson(j: json.JsValue) = NetworKitComputeAttribute(
-    (j \ "op").as[String], (j \ "options").as[json.JsObject])
-  def run(name: String, es: EdgeBundle, options: Map[String, Any] = Map())(
+    (j \ "op").as[String], (j \ "weighted").as[Boolean], (j \ "options").as[json.JsObject])
+  def run(
+    name: String,
+    es: EdgeBundle,
+    options: Map[String, Any] = Map(),
+    weight: Option[Attribute[Double]] = None)(
     implicit
     m: MetaGraphManager): Attribute[Double] = {
-    val op = NetworKitComputeAttribute(name, NetworKitCommon.toJson(options))
+    val op = NetworKitComputeAttribute(name, weight.isDefined, NetworKitCommon.toJson(options))
     import Scripting._
-    op(op.es, es).result.attr
+    weight match {
+      case Some(weight) => op(op.es, es)(op.weight, weight).result.attr
+      case None => op(op.es, es).result.attr
+    }
```

Comment:
Nit: `result.attr` after the `}`?

Reply:
> Nit: `result.attr` after the `}`?

Doesn't work just like that:
```
app/com/lynxanalytics/biggraph/graph_operations/NetworKit.scala:41: ';' expected but '.' found.
    }.result.attr
     ^
one error found
```
If I have to add an extra variable, this little repetition is probably easier on the eyes.

Reply:
Yeah, don't introduce an extra variable for sure. It doesn't really matter, totally fine to leave as is, but maybe it would work if you did `(weight match { ... }).result.attr`?